### PR TITLE
Reduce computation for lbpool states check

### PR DESCRIPTION
**Pull request by mtkh**

### CODE REVIEW

File: src/states_lbpool.py [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+
+def get_lbpools():
+    try:
+        with open("/etc/iglb/lbpools.json") as jsonfile:
+            lbpools_obj = json.load(jsonfile)
+    except IOError:
+        with open('/etc/iglb/iglb.json') as jsonfile:
+            lbpools_obj = json.load(jsonfile)['lbpools']
+    return lbpools_obj
```
Comment (TuxPowered42) [inline src/states_lbpool.py]: Exceptions are for handling errors, this is not an error but rather a configuration option. Just iterate over both files and check if each exist and use open this one.

File: src/states_lbpool.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+import json, subprocess, re
+
+WARNING = 0.70
+CRITICAL = 0.85
+nagios_service = 'states_lbpool'
+
+
+def main():
+    debug = False
```
Comment (TuxPowered42) [inline src/states_lbpool.py]: Please make this a commandline argument. Or see comment about Nagios servers.

File: src/states_lbpool.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+                            default_state_limit,
+                            nagios_service,
+                            separator,
+                            )
+
+    if debug:
+        print("default_state_limit is {}\n".format(default_state_limit))
+        print(send_msg)
+    else:
+        for monitor in ('af-monitor.admin.ig.local', 'aw-monitor.ig.local'):
```
Comment (TuxPowered42) [inline src/states_lbpool.py]: Please make nagios servers configurable from commandline parameters. Even better, lack of Nagios servers could mean displaying output to stdout, that is the debug mode.

File: src/states_lbpool.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+    msg = ''
+    for lbname, lb_params in lbpools.items():
+        statuses = 'States: '
+        exit_code = local_exit_code = 0
+        if lb_params['state_limit']:
+            state_limit = int(lb_params['state_limit'])
+        else:
+            state_limit = default_state_limit
+        lbpool = lb_params['pf_name']
+        if lb_params['default_snat'] != True and lb_params['protocol_port'] and \
```
Comment (TuxPowered42) [inline src/states_lbpool.py]: No need to check booleans for exact True.

File: src/states_lbpool.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+    # The default limit set in pf.conf, fetch it from what is loaded in pf
+    default_limits, err1 = subprocess.Popen(['sudo', 'pfctl', '-sm'],
+                                            stdout=subprocess.PIPE,
+                                            stderr=subprocess.PIPE).communicate()
+
+    states_dict = pfctl_parser(pfctl_output)
+
+    # Hard state limit is always printed in first line
+    default_state_limit_line = default_limits.decode().splitlines()[0]
```
Comment (TuxPowered42) [inline src/states_lbpool.py]: Don't trust which line it is. It might change at any point.

File: src/states_lbpool.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import json, subprocess, re
+
+WARNING = 0.70
+CRITICAL = 0.85
```
Comment (TuxPowered42) [inline src/states_lbpool.py]: This should be configurable from command line arguments.

File: src/states_lbpool.py (hunk 3 above)
Comment (mtkh) [reply src/states_lbpool.py]: done

File: src/states_lbpool.py (hunk 6 above)
Comment (mtkh) [reply src/states_lbpool.py]: done

File: src/lbpool_states.py [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+def main():
+    parser = ArgumentParser()
+    parser.add_argument(
+        '-H', dest='nsca_srv', nargs='+',
+        help='Nagios servers to report the results to'
+    )
+
+    parser.add_argument(
+        '-W', dest='warning', type=int, default=70,
```
Comment (TuxPowered42) [inline src/lbpool_states.py]: Those are lower case parametrs per standard Nagios convenction and full word parameters should be there too, see https://nagios-plugins.org/doc/guidelines.html#AEN302

File: src/lbpool_states.py [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+            try:
+                with open(config) as jsonfile:
+                    dict = json.load(jsonfile)
+                    if 'lbpools' in dict.keys():
+                        lbpools_obj = dict['lbpools']
+                    else:
+                        lbpools_obj = dict
+            except IOError:
+                print("Config file not found")
+                exit(1)
```
Comment (TuxPowered42) [inline src/lbpool_states.py]: Exiting from inside of program is highly discouraged. You should return through normal path to main function if there is any cleanup to be done or just don't catch this exception at all. Your hadler just prints a message and quits anyway.

File: src/lbpool_states.py [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+        exit_code = local_exit_code = 0
+        if lb_params['state_limit']:
+            state_limit = int(lb_params['state_limit'])
+        else:
+            state_limit = default_state_limit
+        lbpool = lb_params['pf_name']
+        critical = state_limit * (crit * 0.01)
+        warning = state_limit * (warn * 0.01)
+
+        if not lb_params['default_snat'] and \
```
Comment (TuxPowered42) [inline src/lbpool_states.py]: You can put all conditions in `()`, then `\` are not necessary.

File: src/lbpool_states.py [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+                lb_params['protocol_port'] and \
+                lb_params['nodes'] and \
+                lbpool in states_dict:  # To exclude those lbpools which have just
+                                        # been made but not deployed
+
+            cur_states = int(states_dict[lbpool]['cur_states'])
+
+            if cur_states >= critical:
+                local_exit_code = 2
+                statuses += 'Used states are above {}% of states limit | Total states limit: {}, Current states: {}'.format(
```
Comment (TuxPowered42) [inline src/lbpool_states.py]: This line is over 80 characters long. Our code must be PEP-8 compliant.

File: src/lbpool_states.py (hunk 7 above)
Comment (mtkh) [reply src/lbpool_states.py]: added

File: src/lbpool_states.py (hunk 8 above)
Comment (mtkh) [reply src/lbpool_states.py]: ok, removed

File: src/lbpool_states.py (hunk 9 above)
Comment (mtkh) [reply src/lbpool_states.py]: I have changed it to some other place. its removed now.

File: src/lbpool_states.py (hunk 10 above)
Comment (mtkh) [reply src/lbpool_states.py]: Corrected, thanks.

**Pull request merged**